### PR TITLE
More improvements

### PR DESCRIPTION
**Pull request by ml31415**
- Added identity API for login, keepAlive etc.
- Better endpoint handling and configuration
- Request objects can generate http headers and body for smooth integration into client implementations 
- Added sample client
- Added live tests against BF servers, given login credentials found in the shell environment
- Working error handling

No need to merge this yet, the tests run, but still WIP. If you had something different in mind in terms of integration with http clients, let me know. Feedback welcome. But in general, I guess we agree, that it should be up to the user, to use this parser with whatever http lib they fancy.

### CODE REVIEW

**Review by limx0 — APPROVED** [review]

**Comment by limx0** [comment]
Awesome! Thanks @ml31415 - happy for this to go in whenever you're finished.

> If you had something different in mind in terms of integration with http clients, let me know. Feedback welcome. But in general, I guess we agree, that it should be up to the user, to use this parser with whatever http lib they fancy.

Yep - totally same page on this. We can include a sample client but happy to also leave that up to the user.

File: betfair_parser/spec/streaming/mcm.py [hunk 1]
```diff
@@ -98,71 +100,26 @@ def event_type_name(self) -> str:
...
-    price: float
-    volume: float
-
-
-class StartingPriceLay(BaseMessage, array_like=True, frozen=True):
-    """StartingPriceLay"""
-
-    price: float
-    volume: float
+AvailableToBack = Annotated[_PriceVolume, msgspec.Meta(title="AvailableToBack")]
```
Comment (limx0) [inline betfair_parser/spec/streaming/mcm.py:114]: 🥳

**Review by limx0 — APPROVED** [review]

**Comment by ml31415** [comment]
If you have some spare empty betfair testing account and feel brave enough to trust the credentials to github, you might set `BF_USERNAME`, `BF_PASSWORD`, a delayed `BF_APPKEY`, and optionally also `BF_CERTIFICATE` and `BF_CERTIFICATE_KEY`. The last two should be file variables. Then we'd have that live tests run in CI as well.

**Comment by limx0** [comment]
Just a couple of little changes: 
- Remove `runner_id` on your advice that it's not an identifier.
- Remove the requests/httpx checks in `test_live.py` - requests is part of the dev deps.
- Added `--check-untyped-defs` in mypy config and fixed some typing issues

**Comment by limx0** [comment]
Merging - amazing work @ml31415 ! Many thanks again.

**Comment by ml31415** [comment]
Hi Brad,

yeah, that EventTypeIdCode is a bit of a weird thing. There is already EventTypeId defined as an Annotated type (as did betfair), so no need to replace it explicitly with IntStr. The EventTypeIdCode I added explicitly for allowing IDE support when constructing requests in your own code. It should not show up anywhere else except in client code. Maybe we should give it another name, to make that more obvious?

**Comment by limx0** [comment]
Ah right, copy - I changed it because mypy was complaining about the different types but maybe I was mixing up their usage.

What you had is fine now that I understand the intention - I'll try and revert it.

**Comment by ml31415** [comment]
Yeah, all that weird type complications already caused me some headache already. Not a friend of squelching the typing errors either but it seems to be the only solution in some cases.

**Comment by limx0** [comment]
With no favours from the Betfair API ..  😢

**Comment by ml31415** [comment]
Yeah, these pointless quotations, `"null"`, `"5.5"` and so on is something that probably would need better handling from `msgspec`. Guess we should open another issue there. But unfortunately, so far I didn't get a response even for the last two issues I had opened there.

**Comment by limx0** [comment]
Yeah its a tough one - msgspec relies on the JSON spec I think, and I wouldn't expect it to handle the malformed data Betfair is spitting out, we may need to just handle it on our side.

I've found `jcrist` (msgspec author) to be quite responsive and helpful in the past, he may just be busy at the moment, lets just keep an eye on the issues.

**Comment by ml31415** [comment]
Yeah, I saw that. Hope he'll find the time to have a look at that. I'll post an issue report in any case later.

**Pull request merged**